### ISSUE TITLE
Add support for ALTER TYPE .. { ADD | DROP | ALTER |RENAME } ATTRIBUTE

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#15518]
Oracle and PostgreSQL allow for mutating UDT definitions by supporting the following operations on `ATTRIBUTE`:

- [ ] `ADD`
- [ ] `ALTER`
- [ ] `DROP`

See: 
- https://www.postgresql.org/docs/current/sql-altertype.html
- https://docs.oracle.com/en/database/oracle/oracle-database/23/lnpls/ALTER-TYPE-statement.html#GUID-A8B449E7-E3A8-48F4-A4C6-5BB87B1841CD

Now that we support DDL on `TYPE` objects (see https://github.com/jOOQ/jOOQ/issues/8474), we should offer this functionality as well

**Comment by Fridious** [comment]
Hey,
I'm writing to inquire about the status of the feature request. I'm interested in knowing whether work has already begun on this feature and what the current status is?
Additionally, I'm quite enthusiastic about contributing to the project and would like to offer to implement this feature. If there's any guidance you could provide or specific aspects I should pay attention to, please let me know.

**Comment by lukaseder** [comment]
> I'm writing to inquire about the status of the feature request. I'm interested in knowing whether work has already begun on this feature and what the current status is?

This doesn't look like a high priority issue, though upvoting it always helps for a better ranking here:
https://github.com/jOOQ/jOOQ/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

> Additionally, I'm quite enthusiastic about contributing to the project and would like to offer to implement this feature. If there's any guidance you could provide or specific aspects I should pay attention to, please let me know.

It's unlikely this is the right approach, see our stance on contributions:
https://github.com/jOOQ/jOOQ/blob/main/CONTRIBUTING.md

**Comment by Fridious** [comment]
@lukaseder I understand, I'll begin by creating the API and implementing it. I will submit a PR for the API first, so you could use this at least. If my implementation code aligns with the established concept, you can also use it. In case it doesn't match and an official implementation is available in the future, I can easily replace my code with your implementation. Raw SQL statements are not suitable for me."

**Comment by lukaseder** [comment]
> I'll begin by creating the API and implementing it. I will submit a PR for the API first, so you could use this at least. If my implementation code aligns with the established concept, you can also use it. In case it doesn't match and an official implementation is available in the future, I can easily replace my code with your implementation.

I can tell you already now that I'll reject any PR for this. Have you read the document here?
https://github.com/jOOQ/jOOQ/blob/main/CONTRIBUTING.md

Of course, you can still do this on your end, for yourself, but I don't see 1) why your own implementation and PR should change prioritisation of this improvement and 2) why time should be spent reviewing and explaining, given that this feature isn't currently being prioritised.

> Raw SQL statements are not suitable for me.

You can wrap this statement in a utility that uses plain SQL templating. I wouldn't overengineer this, but I won't stop you if you must :) Just beware that I will not allocate time to review any PR, nor will I look at it for the actual implementation.

**Comment by Fridious** [comment]
@lukaseder yes I read. This would also be a good approach for me to just create a utility and maybe try to use the internal SQL templating engine of this project.

**Comment by lukaseder** [comment]
Great. Good luck then! :)
